runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -267,26 +267,39 @@ public ProjectionFunction getProjectionFunction() {
         }
     }
 
-    /** The "main" map which contains the primary key-value mappings. */
-    private final Map<K,V> mainMap = new HashMap<>();
+    // The "main" map which contains the primary key-value mappings.
+    private final Map<K,V> mainMap;
     private Set<Index<K, V, ? extends Comparable>> indexSpec = new HashSet<>();
     private final Map<String, Map<Comparable, Map<K, V>>> secondaryIndexes = new HashMap<>();
 
     @Getter
     boolean indexGenerationFailed = false;
 
-    /** Generate a table with the given set of indexes. */
-    public CorfuTable(IndexRegistry<K, V> indices) {
+    /**
+     * Generate a table with a given implementation for the mainMap
+     */
+    public CorfuTable(IndexRegistry<K, V> indices, Map<K, V> mapImpl) {
```

Comment:
It might be cleaner to pass a type and not an instance. 

https://stackoverflow.com/questions/3574065/instantiate-a-class-object-with-constructor-that-accepts-a-string-parameter

Reply:
I thought about this, but was wondering about passing custom implementation of the map. For example if we pass only a LinkedHashMap or HashMap as a type and have CorfuTable instantiate it, that will limit us from passing in some other Map which can do other things. Any thoughts?